components/compliance-service/api/reporting/server/server.go

```diff
@@ -383,38 +382,48 @@ func formatFilters(filters []*reporting.ListFilter) map[string][]string {
 	return formattedFilters
 }
 
-func validatePaginationAndSorting(in *reporting.Query, validSortFields map[string]string, defaultSortField string) (from int32, per_page int32, sort string, asc bool, err error) {
+type pageSortInfo struct {
+	from    int32
+	perPage int32
+	sort    string
+	asc     bool
+}
+
+func validatePaginationAndSorting(in *reporting.Query, validSortFields map[string]string, defaultSortField string) (pageSortInfo, error) {
+	ret := pageSortInfo{}
+
 	if in.PerPage == 0 {
 		in.PerPage = 10
 	}
-	per_page = in.PerPage
+	ret.perPage = in.PerPage
 
 	if in.Page == 0 {
 		in.Page = 1
 	}
-	from = (in.Page - 1) * in.PerPage
+	ret.from = (in.Page - 1) * in.PerPage
 
-	switch order := in.Order; order {
-	case 0:
-		asc = true
-	case 1:
-		asc = false
+	switch in.Order {
+	case reporting.Query_ASC:
+		ret.asc = true
+	case reporting.Query_DESC:
+		ret.asc = false
 	}
+
+	sort := defaultSortField
 	if in.Sort != "" {
-		valid_fields := make([]string, 0)
-		for k := range validSortFields {
-			valid_fields = append(valid_fields, k)
-		}
-		if !stringutils.SliceContains(valid_fields, in.Sort) {
-			sorter.Strings(valid_fields)
-			err = &errorutils.InvalidError{Msg: fmt.Sprintf("Parameter 'sort' only supports one of the following fields: %v", valid_fields)}
-			return
+		if _, ok := validSortFields[in.Sort]; !ok {
+			validFields := make([]string, 0, len(validSortFields))
+			for k := range validSortFields {
+				validFields = append(validFields, k)
+			}
+			sorter.Strings(validFields)
+			return ret, &errorutils.InvalidError{Msg: fmt.Sprintf("Parameter 'sort' only supports one of the following fields: %v", validFields)}
 		}
 		sort = in.Sort
-	} else {
-		sort = defaultSortField
 	}
-	return
+	ret.sort = sort
```

Comment:
I know it's not really your job to make this squeaky clean (thanks though!), but this could probably be the value of the sort field map rather than the key. Does `ret.sort` get used for anything other than looking up the value in the map?

Reply:
I like this suggestion and since this isn't urgent I went ahead and did it.